File: 0x08-python-more_classes/4-rectangle.py
#!/usr/bin/python3


class Rectangle:
    def __init__(self, width=0, height=0):
        self.width = width
        self.height = height

    @property
    def width(self):
        return self.__width

    @width.setter
    def width(self, value):
        if type(value) is not int:
            raise TypeError("width must be an integer")
        elif value < 0:
            raise ValueError("width must be >= 0")
        else:
            self.__width = value

    @property
    def height(self):
        return self.__height

    @height.setter
    def height(self, value):
        if type(value) is not int:
            raise TypeError("height must be an integer")
        elif value < 0:
            raise ValueError("height must be >= 0")
        else:
            self.__height = value

    def area(self):
        return self.__height * self.__width

    def perimeter(self):
        if self.__height == 0 or self.__width == 0:
            return 0
        return self.height * 2 + self.__width * 2

    def __str__(self):
        s = ""
        if self.__height == 0 or self.__width == 0:
            return s
        for x in range(self.__height):
            s += self.__width * "#" + "\n"
        return s[:-1]

    def __repr__(self):
        return "Rectangle(" + str(self.width) + ", " + str(self.height) + ")"

File: 0x12-javascript-web_scraping/4-rectangle.js
#!/usr/bin/node
exports.Rectangle = function Rectangle (w, h) {
  if (w > 0 && h > 0 && w !== undefined && h !== undefined) {
    this.width = w;
    this.height = h;
  }
  this.print = function () {
    for (let i = 0; this.height > i; ++i) {
      console.log('X'.repeat(this.width));
    }
  };
  this.rotate = function () {
    const tmp = this.width;
    this.width = this.height;
    this.height = tmp;
  };
  this.double = function () {
    this.width *= 2;
    this.height *= 2;
  };
};
